src/mlx90632_extended_meas.c

```diff
@@ -316,17 +345,25 @@ int32_t mlx90632_get_meas_type(void)
 {
     int32_t ret;
     uint16_t reg_ctrl;
+    uint16_t reg_temp;
 
-    ret = mlx90632_i2c_read(MLX90632_REG_CTRL, &reg_ctrl);
+    ret = mlx90632_i2c_read(MLX90632_REG_CTRL, &reg_temp);
     if (ret < 0)
         return ret;
 
-    reg_ctrl = reg_ctrl & MLX90632_CFG_MTYP_MASK;
+    reg_ctrl = reg_temp & MLX90632_CFG_MTYP_MASK;
     reg_ctrl = reg_ctrl >> 4;
 
     if ((reg_ctrl != MLX90632_MTYP_MEDICAL) & (reg_ctrl != MLX90632_MTYP_EXTENDED))
         return -EINVAL;
 
+    reg_temp = reg_temp & MLX90632_CFG_PWR_MASK;
+
+    if (reg_temp == MLX90632_PWR_STATUS_SLEEP_STEP)
+        return reg_ctrl + 128;
```

Comment:
Should also be a macro explaining that high bit is actually only for software purposes and that bottom is used for the masking.